modules/channel-messenger/src/backend/messenger.ts

```diff
@@ -0,0 +1,165 @@
+      return
+    }
+
+    for (const entry of body.entry) {
+      // Will only ever contain one message, so we get index 0
+      const webhookEvent = entry.messaging[0]
+      const senderId = webhookEvent.sender.id
+
+      if (webhookEvent.message) {
+        await messenger.handleMessage(senderId, webhookEvent.message)
+      } else if (webhookEvent.postback) {
+        await messenger.handleMessage(senderId, { text: webhookEvent.postback.payload })
+      }
+    }
+
+    res.status(200).send('EVENT_RECEIVED')
+  }
+
+  async _setupWebhook(req, res) {
+    const mode = req.query['hub.mode']
+    const token = req.query['hub.verify_token']
+    const challenge = req.query['hub.challenge']
+    const botId = req.params.botId
+
+    const messenger = await this.forBot(botId)
+    const config = await messenger.getConfig()
+
+    if (mode && token && mode === 'subscribe' && token === config.verifyToken) {
+      this.bp.logger.forBot(botId).debug('Webhook Verified.')
+      res.status(200).send(challenge)
+    } else {
+      res.sendStatus(403)
+    }
+
+    await messenger.setupGreeting()
+    await messenger.setupGetStarted()
+  }
+}
+
+export class ScopedMessengerService {
+  private config: Config
+
+  constructor(private botId: string, private bp: typeof sdk, private http: AxiosInstance) {}
+
+  async handleMessage(senderId: string, message) {
+    this.config = await this.getConfig()
+
+    if (message.text) {
+      const content = await this.bp.converse.sendMessage(this.botId, senderId, message, 'messenger')
+
+      // Responses are split into pairs {typing, message}
+      for (let i = 0; i < content.responses.length; i += 2) {
+        const isTyping = content.responses[i].value
+        const message = content.responses[i + 1]
+
+        isTyping && (await this._sendAction(senderId, 'typing_on'))
+        await this._sendMessage(senderId, message)
+        isTyping && (await this._sendAction(senderId, 'typing_off'))
+      }
+    }
+  }
+
+  async getConfig(): Promise<any> {
+    if (this.config) {
+      return this.config
+    }
+
+    return await this.bp.config.getModuleConfigForBot('channel-messenger', this.botId)
+  }
+
+  async setupGetStarted(): Promise<void> {
+    const message = this.config.getStarted
+    if (!message) {
+      return
+    }
+
+    const body = {
+      get_started: {
+        payload: message
+      }
+    }
+
+    await this._sendProfile(body)
+  }
+
+  async setupGreeting(): Promise<void> {
+    const message = this.config.greeting
+    if (!message) {
+      return
+    }
+
+    const payload = {
+      greeting: [
+        {
+          locale: 'default',
+          text: message
+        }
+      ]
+    }
+
+    await this._sendProfile(payload)
+  }
+
+  private async _sendAction(psid, action) {
+    const body = {
+      recipient: {
+        id: psid
+      },
+      sender_action: action
+    }
+
+    await this.http.post(`/messages`, body, { params: { access_token: this.config.verifyToken } })
```

Comment:
How do you look to move `{ params: { access_token: this.config.verifyToken } }` to `this.FBRequestConfig`